src/components/_tests_/EnumSwitch.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { mount, shallow } from "enzyme";
+import { createElement } from "react";
+import * as classNames from "classnames";
+import { Alert, AlertProps } from "../Alert";
+
+import { EnumSwitch, EnumSwitchProps } from "../EnumSwitch";
+import { EnumButton } from "../EnumButton";
+
+describe("EnumSwitch", () => {
+    const shallowRenderSwitch = (props: EnumSwitchProps) => shallow(createElement(EnumSwitch, props));
+    const fullRenderSwitch = (props: EnumSwitchProps) => mount(createElement(EnumSwitch, props));
+
+    const defaultProps: EnumSwitchProps = {
+        alertMessage: "",
+        enumList: [],
+        enumAttributeValue: "Chicken",
+        onClickAction: jasmine.createSpy("onClick"),
+        status: "enabled",
+        bootstrapStyle: "default"
+    };
+
+    it("renders the structure correctly", () => {
+        const enumswitch = shallowRenderSwitch(defaultProps);
+
+        expect(enumswitch).toBeElement(
+            createElement("div", { className: "form-validation" },
+                createElement("div", { className: "widget-enum-switch form-control" },
+                    createElement(EnumButton, {
+                        status: defaultProps.status,
+                        bootstrapStyle: defaultProps.bootstrapStyle,
+                        position: enumswitch.state().position,
+                        visibility: enumswitch.state().visibility,
+                        width: enumswitch.state().width,
+                        height: enumswitch.state().height,
+                        getButtonNode: jasmine.any(Function)
+                    })
+                ),
+                createElement(Alert, { message: defaultProps.alertMessage || "", bootstrapStyle: "danger" })
+            ));
+    });
+
+    it("should not create button if there is no default atrribute", () => {
+        defaultProps.enumAttributeValue = "";
```

Comment:
If you are sharing variables in unit tests. it should not change. If it changes create a copy and use in your tests